include/oxen/quic/loop.hpp

```diff
@@ -22,6 +22,45 @@ namespace oxen::quic
 
     static void setup_libevent_logging();
 
+    using loop_time = std::chrono::microseconds;
+    using loop_ptr = std::shared_ptr<::event_base>;
+
+    class Loop;
+
+    struct EventHandler : public std::enable_shared_from_this<EventHandler>
+    {
+        friend class Loop;
+
+      private:
+        std::atomic<bool> _is_running{false};
+        bool _is_stopped{false};
+        event_ptr ev;
+        timeval interval;
+        std::function<void()> f;
+
+        void start_event(
+                const loop_ptr& _loop,
+                loop_time _interval,
+                std::function<void()> task,
+                bool persist = true,
+                bool start_immediately = true);
+
+        EventHandler() = default;
+
+      public:
+        ~EventHandler();
+
+        bool is_running() const { return _is_running and !_is_stopped; }
+
+        bool is_paused() const { return !(_is_running or _is_stopped); }
+
+        bool is_stopped() const { return _is_stopped; }
+
+        bool start();
+        bool pause();
+        bool stop();
```

Comment:
`stop()` seems unnecessary: its implementation is effectively the same as the destructor, which means an object on which you have called `stop()` is a sort of unusable zombie object.  It would be preferable to use the RAII pattern and have the destructor be the (only) permanent stop.  Then `pause()` can get renamed to `stop()` (which is a nicer linguistic inverse of `start()`), and you simplify the state tracking to just one atomic bool.